src/Features/LanguageServer/Protocol/CapabilityRegistrationsProviderFactory.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer
+{
+    [ExportCSharpVisualBasicLspServiceFactory(typeof(ICapabilityRegistrationsProvider)), Shared]
+    internal class CapabilityRegistrationsProviderFactory : ILspServiceFactory
+    {
+        private readonly IEnumerable<Registration> _registrations;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CapabilityRegistrationsProviderFactory([ImportMany] IEnumerable<Registration> registrations)
```

Comment:
Maybe I'm misunderstanding how the MEF scopes work with this, but importing (and exporting) VS Protocol types seems odd to me, especially with respect to my other comment, of suppliying client apabilities to influence the server capabilities. Perhaps Xaml (and Razor) should just export this service, and the server allow for multiple?

Reply:
Hmm, @dibarbet what do you think about this suggestion? The reason there's only one now is because `ICapabilityRegistrationsProvider` is an `ILspService`, and importing many exported registration was rather clean and simple.

Reply:
Catching up.  I still think we want to approach this entire bit differently.

I think we don't need this type at all.  Just have XAML implement an IOnInitialized LSP service that sends the registrations it wants.  I don't think we need to define an explicit API in the language server to allow that.